### ISSUE TITLE
Is there a signed version of SAR?

### ISSUE THREAD

**Issue by omerdn** [eiz/SynchronousAudioRouter#116]
I dont want launch the testsigned mode on. I can use the old versions etc. Is there any version of signed on windows?

**Comment by 19Topgun93** [comment]
I found out, that Windows 10 can do this know. I have looking after 3 programs, but it is so easy with windows 10 https://www.youtube.com/watch?v=PuN9EPM5zgg&list=PLB-XE8Gvu0g6o9kCUrseBpCW3x_2Szffh&index=6

**Comment by HGhostRevisionist88** [comment]
While 19Topogun93 video explains how Windows can be used for some basic audio routing, I don't think it can be compared to what SAR would offer with how ASIO offers many channels to route if necessary, and the problem is that to make things similar would have to use Virtual Audio Cable or VB-Audio Cables/Voicemeeter to be able to do it. I would still say that the problem is how you mix the audio, especially Microphones since Windows audio mixer is basic, eartrumpet which is better doesn't offer to adjust mic either.

And the problem I see as well is how DAWs and most programs will only allow one input and output device to be assigned so while you can achieve something like Asio4all + SAR with VAC and the Audio Repeater, it would get too complicated to be worth it. But it you can create 100000 virtual audio cables you can only use 1 or 2 directly in a Daw which sucks now WASAPI is a good alternative to ASIO, especially if you use Asio4all.
I know, there is Voicemeeter which makes things easier, but I don't like how it performs and the CPU usage of Banana and Potato is higher than using Studio One with some VSTs (and Voicemeeter doesn't even support VSTs), so that's why I bought VAC.

Of course then I found SAR and even SoundMixr (https://github.com/KaixoCode/SoundMixr) that is designed to use it mainly with SAR if needed, without going into a 'complicated' full featured DAW like Ardour which is free or Reaper or Studio One which I use for everything even to route my normal PC audio.

Anyway. while I like the idea of SAR and being free and all, it's easy to understand why some people wouldn't use this latest version since drivers aren't signed.
If interested, I would recommend Asio Link Pro, it if you want to try it and you don't want to wait for SAR to sign the drivers etc. The thing is Asio Link Pro became free when the developer died 3 years ago and some people reverse engineered and patch it, supposedly the cousin agreed to that so it is not like an illegal thing or anything, but it means that if a Windows Update or something breaks it, then nobody will be there to fix it. But Asio Link is still the best if you need a program that "_Route audio to/from WDM/MME/DirectSound/WASAPI and LAN/NET_" or can be used in more than 1 app at the same time, or if you need to route audio and you don't want to use a DAW or something since it work as standalone if needed.  There is even an option in the installer to install 16 virtual audio cables and not just 4, if you need that. But it seems like it is the most powerful windows app for that even after 3 years from last update.
Of course I like what SAR offers and keep checking to see when I can truly tested without having to restart my computer and use the testing mode (I dont't like the BCDedit stuff) and I like VAC too (wasn't cheap but oh well), but Asio Link Pro makes things great and easy so that's why I would still recommend it today.

Of course hope SAR grows and matures more, especially if Asio Link Pro breaks eventually but if I wanted to mention also an alternative in case people want to try something else that works greats too.

**Comment by 19Topgun93** [comment]
> While 19Topogun93 video explains how Windows can be used for some basic audio routing, I don't think it can be compared to what SAR would offer with how ASIO offers many channels to route if necessary, and the problem is that to make things similar would have to use Virtual Audio Cable or VB-Audio Cables/Voicemeeter to be able to do it. I would still say that the problem is how you mix the audio, especially Microphones since Windows audio mixer is basic, eartrumpet which is better doesn't offer to adjust mic either.
> 
> And the problem I see as well is how DAWs and most programs will only allow one input and output device to be assigned so while you can achieve something like Asio4all + SAR with VAC and the Audio Repeater, it would get too complicated to be worth it. But it you can create 100000 virtual audio cables you can only use 1 or 2 directly in a Daw which sucks now WASAPI is a good alternative to ASIO, especially if you use Asio4all.
> I know, there is Voicemeeter which makes things easier, but I don't like how it performs and the CPU usage of Banana and Potato is higher than using Studio One with some VSTs (and Voicemeeter doesn't even support VSTs), so that's why I bought VAC.
> 
> Of course then I found SAR and even SoundMixr (https://github.com/KaixoCode/SoundMixr) that is designed to use it mainly with SAR if needed, without going into a 'complicated' full featured DAW like Ardour which is free or Reaper or Studio One which I use for everything even to route my normal PC audio.
> 
> Anyway. while I like the idea of SAR and being free and all, it's easy to understand why some people wouldn't use this latest version since drivers aren't signed.
> If interested, I would recommend Asio Link Pro, it if you want to try it and you don't want to wait for SAR to sign the drivers etc. The thing is Asio Link Pro became free when the developer died 3 years ago and some people reverse engineered and patch it, supposedly the cousin agreed to that so it is not like an illegal thing or anything, but it means that if a Windows Update or something breaks it, then nobody will be there to fix it. But Asio Link is still the best if you need a program that "_Route audio to/from WDM/MME/DirectSound/WASAPI and LAN/NET_" or can be used in more than 1 app at the same time, or if you need to route audio and you don't want to use a DAW or something since it work as standalone if needed. There is even an option in the installer to install 16 virtual audio cables and not just 4, if you need that. But it seems like it is the most powerful windows app for that even after 3 years from last update.
> Of course I like what SAR offers and keep checking to see when I can truly tested without having to restart my computer and use the testing mode (I dont't like the BCDedit stuff) and I like VAC too (wasn't cheap but oh well), but Asio Link Pro makes things great and easy so that's why I would still recommend it today.
> 
> Of course hope SAR grows and matures more, especially if Asio Link Pro breaks eventually but if I wanted to mention also an alternative in case people want to try something else that works greats too.

Yeah I agree. I have now th GOXLR from TC Helion. Win 10 can do the basic route with the app routing, and mei GOXLR do the other work. With this I have 4 output device. With windows I route it to the 4 Output, because, if I don't do this, I have the problem, if the game or the programm has no choice to configure the audio. In that case, I have all at one Audio output in the worst case :) 

So, the Windows Routing is is a little helper vor other software kike SAR, I think.

**Comment by HGhostRevisionist88** [comment]
> Yeah I agree. I have now th GOXLR from TC Helion. Win 10 can do the basic route with the app routing, and mei GOXLR do the other work. With this I have 4 output device. With windows I route it to the 4 Output, because, if I don't do this, I have the problem, if the game or the programm has no choice to configure the audio. In that case, I have all at one Audio output in the worst case :)
> 
> So, the Windows Routing is is a little helper vor other software kike SAR, I think.

In that case, since you really have the GOXLR I hope you are using EarTrumpet (https://github.com/File-New-Project/EarTrumpet), since it makes it easier to route the apps by just right clicking on them and selecting the device or drag and dropping them to another device when you expand the whole list of audio devices. Easier than opening the sound settings just for that and then scroll down to go to the advanced sound options etc.
Something like EarTrumpet is what you would expect Microsoft to replace the volume mixer with, of course volume tray icon became a tiny bit better when they updated it to make it easier to switch output devices, and it even got a similar look to EarTrumpet but it still misses the other features EarTrumpet offers.

**Comment by omerdn** [comment]
I actually wanted to just use signed version so i use the old version and it doesnt gives me any problem actually. But i wasnt know the soundmixr and i actually liked it thanks for the giving information.

Asio Link Pro really bad working, and i'm also sad for its developer

I just want to know now, can i use soundmixr with and old version of SAR?

**Comment by HGhostRevisionist88** [comment]
Yes, it does and It should work with any ASIO driver because all it does is get the ASIO channels on the mixer, but I guess the "_but specifically made for SAR_" is about how it gets the names and channels and stuff from SAR unlike with other ASIO drivers.

And that sucks about Asio Link Pro, what's wrong with it though? I know it has to do with preference and necessity as well, so it can be so much for what someone needs but I still think it is the best windows program of its class even if GUI looks like from 90s and it looks a little confusing when you first start it. But I know it is also about how it works really good with Studio One, because in DAWs like Reaper you get the 64 channels of the asio4all driver for example, while on studio one I just get the channels I set up in Studio one so it becomes less confusing to route and know where things are doing and going to.

Anyway! I am glad SoundMixr can be useful for your needs, it even has some effects which is cool so no big complicated software with VST is needed for that and it is easy to route things and it seemed stable enough when I did some tests with it and SAR. So a more basic non-complicated setup is always useful and good! 👌

**Comment by vassilyzaitsev** [comment]
@HGhostRevisionist88 OMG, thank you so much to indirectly introduce me to this ASIO Link Pro. SAR was already working better than Voicemeeter stuff (with a few glitches/clicks). But now, everything is running smoothly on my 2012 laptop + Audiobox 44vsl. Amazing. ❤

![image](https://user-images.githubusercontent.com/35409978/119656647-c6775c80-bddf-11eb-96bf-7cb21d9d57b8.png)

**Comment by cdhowie** [comment]
This is a duplicate of #86.

I second the suggestion to use Asio Link Pro instead. It is working very well for me and, in fact, I have far fewer xruns using it than I can achieve with SAR.  (Which, perhaps counterintuitively, is rather unfortunate since SAR is more likely to work into the future.)

**Comment by musicalman** [comment]
Hi all,
A friend told me about sar today and I wanted to try it. Unfortunately that is proving difficult because of this signing issue.

To start I should probably preface this by saying that I am blind, so I have to use screen reading software to use a computer. This will become more significant in a bit.

I tried installing the latest version of sar, but it outright won't install. The older version 0.13.1 installs, but in device manager, the driver status is unverified. I tried getting it to work in Reaper, probably spending way more time on it than I really should have, but to no avail.

I tried to enable testsigning, but secure boot is enabled and preventing me from making this change. In order to disable that, I have to go into the bios, and that is something screen reading programs can't help with. Thus I'll have to ask a sighted person to help me navigate the bios, but that will have to wait until I find a monitor (I've never really needed to hook a monitor to this PC tbh, and I guess I'm paying for that negligence now).

So yeah... this is overall a frustrating experience. I'm willing to change things in the bios if that's what it takes, but I'm not really thrilled about it either, especially since it's not something I can undo easily if things go wrong. Not that I immediately anticipate something to go wrong, but I do worry, especially when messing with the bios and having to drag someone else into that with me. Lol

So is that my only option, or is there some easier way perhaps to get an older version working?

Thanks for any help!

**Comment by Dirkaios** [comment]
I've been using ASIO Link Pro for some time now.
It is a very good software, but the developer passed away and there will be no more updates.
Also, it still has a bug with audio cutting around the first 100ms  for small audio (very noticeable on windows speakers test).
I successfully tested most recent version of SAR on testing mode.
The thing is: I use it mainly for gaming, testing, chat and creating separate devices.
One for having surround sound + equalizer APO and Hesuvi. Other for plain stereo. Other for Dolby Atmos for Headphones or Windows Sonic and so on.
I think SAR could easily replace ASIO Link Pro in the future and I hope there is some development for that. It does look quite promising, but we really need to be able to use it without having to enable "testing mode" in Windows.
